### PR TITLE
provider/aws: Check ElastiCache node status before returning

### PR DESCRIPTION
**Pull request by catsby**
Check every node for `available` status before returning `available` for the cluster.

Probably addresses #2051

### CODE REVIEW

File: builtin/providers/aws/resource_aws_elasticache_cluster.go [hunk 1]
```diff
@@ -336,6 +337,13 @@ func CacheClusterStateRefreshFunc(conn *elasticache.ElastiCache, clusterID, give
 
 		// return given state if it's not in pending
 		if givenState != "" {
+			// loop the nodes and check their status as well
+			for _, n := range c.CacheNodes {
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_elasticache_cluster.go]: This range will not cover the case where `len(c.CacheNodes) < len(c.NumCacheNodes)`, which is I believe the problematic scenario, yeah?

Comment (catsby) [inline builtin/providers/aws/resource_aws_elasticache_cluster.go]: I suppose in theory that could happen. In my extensive testing of multiple regions, node counts, and types (redis/memcached), I've not seen this happen. But I'll add code to check that as well

**Comment by catsby** [comment]
@phinze updated

**Comment by phinze** [comment]
LGTM!

**Pull request merged**